File: 0733-flood-fill/2T1/0733-flood-fill.js
var floodFill = function(image, sr, sc, color) {
    if(image[sr][sc] == color) return image;
    fill(image, sr, sc, color, image[sr][sc]);
    return image;
}

var fill = function(image, sr, sc, color, old) {
    if(sr >= 0 && sc >= 0 && sr < image.length && sc < image[sr].length && old == image[sr][sc]){
        image[sr][sc] = color;

        fill(image, sr-1, sc, color, old);
        fill(image, sr+1, sc, color, old);
        fill(image, sr, sc-1, color, old);
        fill(image, sr, sc+1, color, old);
    }
};

File: 0733-flood-fill/0733-flood-fill.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int sample = image[sr][sc];
        
        if (sample != color) {
            change(image, sr, sc, color, sample);
        }
        return image;
    }
    
    public void change(int[][] image, int sr, int sc, int color, int sample) {
        if (image[sr][sc] == sample) {
            image[sr][sc] = color;
            if (sr-1 >= 0) change(image, sr-1, sc, color, sample);
            if (sc-1 >= 0) change(image, sr, sc-1, color, sample);
            if (sr+1 < image.length) change(image, sr+1, sc, color, sample);
            if (sc+1 < image[sr].length) change(image, sr, sc+1, color, sample);
        }
    } 
}